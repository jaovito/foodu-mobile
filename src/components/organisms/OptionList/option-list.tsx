import {OptionCard} from '@components/molecules/OptionCard';
import React from 'react';
import {OptionListProps} from './option-list.props';
import {Container, Separator} from './option-list.styles';

export const OptionList = ({options, ...rest}: OptionListProps) => {
  return (
    <Container
      data={options}
      renderItem={({item}) => <OptionCard {...item} />}
      keyExtractor={item => item.name}
      style={rest.style}
      ItemSeparatorComponent={() => <Separator />}
      {...rest.listProps}
    />
  );
};
