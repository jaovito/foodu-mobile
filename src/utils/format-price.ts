import {LanguagePaths} from '@translations/enumns/LanguagePaths';
import {translate} from '@translations/index';

const defaultValues = {
  language: 'en-US',
  currency: 'USD',
};

function getDefaultValues() {
  if (translate(LanguagePaths.GENERAL_LANGUAGE) === 'general.language') {
    return defaultValues;
  }

  return {
    language: translate(LanguagePaths.GENERAL_LANGUAGE),
    currency: translate(LanguagePaths.GENERAL_CURRENCY),
  };
}

export function formatPrice(price: number) {
  return price.toLocaleString(getDefaultValues().language, {
    style: 'currency',
    currency: getDefaultValues().currency,
  });
}
