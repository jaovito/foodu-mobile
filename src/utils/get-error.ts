import {AxiosError} from 'axios';

export function getError(err: Error | unknown): AxiosError {
  if (err instanceof AxiosError) {
    return err;
  }

  return err as AxiosError;
}
