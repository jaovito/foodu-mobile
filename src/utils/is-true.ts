/**
 *
 * Convert booleans values, considering the cases that I understand how true and false.
 * Exemple: Empty arrays and objects, we consider false.
 *
 * @param  param - Can be any value
 *
 * @returns if `value` it isn't an empty array, returns true
 * @returns if `value` it isn't an empty object, returns true
 * @returns if `value` it isn't an empty string, returns true
 * @returns if `value` it isn't null, returns true
 * @returns if `value` it's a non-zero number, returns true
 */
export const isTrue = (param: any): boolean => {
  if (typeof param === 'object') {
    if (param === null) {
      return false;
    } else if (param instanceof Array) {
      return param.length === 0 ? false : true;
    } else {
      return Object.entries(param).length === 0 ? false : true;
    }
  }

  return !!param;
};
