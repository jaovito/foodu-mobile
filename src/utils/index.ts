import {isTrue} from './is-true';

export {isTrue};
