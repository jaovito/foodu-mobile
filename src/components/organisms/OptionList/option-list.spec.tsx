import {mock} from '@test/mocks';
import {render} from '@test/utils/test-utils';
import {OptionList} from './option-list';

describe('<OptionList />', () => {
  it('should be defined', () => {
    const optionList = render(<OptionList options={mock.optionList.options} />);

    expect(optionList).toBeDefined();
  });

  it('should be able to find some list', async () => {
    const {findByText} = render(
      <OptionList options={mock.optionList.options} />,
    );

    const text = await findByText(mock.optionList.options[0].name);

    expect(text).toBeTruthy();
  });

  it('should match with snapshot', () => {
    const optionList = render(
      <OptionList options={mock.optionList.options} />,
    ).toJSON();

    expect(optionList).toMatchSnapshot();
  });
});
