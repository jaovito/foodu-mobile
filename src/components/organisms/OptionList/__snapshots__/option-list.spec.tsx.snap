// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionList /> should match with snapshot 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  data={
    [
      {
        "icon": "map-pin",
        "name": "Location",
      },
      {
        "icon": "user",
        "name": "Profile",
      },
      {
        "icon": "settings",
        "name": "Settings",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  style={
    [
      {
        "width": "100%",
      },
    ]
  }
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <RNGestureHandlerButton
        activeOpacity={0.105}
        collapsable={false}
        delayLongPress={600}
        handlerTag={7}
        handlerType="NativeViewGestureHandler"
        onActiveStateChange={[Function]}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#F0F0F5",
            "borderRadius": 16,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": "100%",
          }
        }
        underlayColor="black"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "black",
              "borderBottomLeftRadius": undefined,
              "borderBottomRightRadius": undefined,
              "borderRadius": 16,
              "borderTopLeftRadius": undefined,
              "borderTopRightRadius": undefined,
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 12,
                },
                [
                  {
                    "bottom": 3,
                    "color": "#0D0D0D",
                    "fontSize": 31,
                    "marginRight": 16,
                  },
                ],
                {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              [
                {
                  "color": "#0D0D0D",
                  "fontFamily": "Poppins_400Regular",
                  "fontSize": 31,
                },
                {
                  "fontSize": 31,
                },
              ]
            }
          >
            Location
          </Text>
        </View>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              [
                {
                  "color": "#0D0D0D",
                  "fontSize": 35,
                },
              ],
              {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </RNGestureHandlerButton>
      <View
        style={
          [
            {
              "marginTop": 16,
            },
          ]
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <RNGestureHandlerButton
        activeOpacity={0.105}
        collapsable={false}
        delayLongPress={600}
        handlerTag={8}
        handlerType="NativeViewGestureHandler"
        onActiveStateChange={[Function]}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#F0F0F5",
            "borderRadius": 16,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": "100%",
          }
        }
        underlayColor="black"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "black",
              "borderBottomLeftRadius": undefined,
              "borderBottomRightRadius": undefined,
              "borderRadius": 16,
              "borderTopLeftRadius": undefined,
              "borderTopRightRadius": undefined,
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 12,
                },
                [
                  {
                    "bottom": 3,
                    "color": "#0D0D0D",
                    "fontSize": 31,
                    "marginRight": 16,
                  },
                ],
                {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              [
                {
                  "color": "#0D0D0D",
                  "fontFamily": "Poppins_400Regular",
                  "fontSize": 31,
                },
                {
                  "fontSize": 31,
                },
              ]
            }
          >
            Profile
          </Text>
        </View>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              [
                {
                  "color": "#0D0D0D",
                  "fontSize": 35,
                },
              ],
              {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </RNGestureHandlerButton>
      <View
        style={
          [
            {
              "marginTop": 16,
            },
          ]
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <RNGestureHandlerButton
        activeOpacity={0.105}
        collapsable={false}
        delayLongPress={600}
        handlerTag={9}
        handlerType="NativeViewGestureHandler"
        onActiveStateChange={[Function]}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#F0F0F5",
            "borderRadius": 16,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": "100%",
          }
        }
        underlayColor="black"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "black",
              "borderBottomLeftRadius": undefined,
              "borderBottomRightRadius": undefined,
              "borderRadius": 16,
              "borderTopLeftRadius": undefined,
              "borderTopRightRadius": undefined,
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 12,
                },
                [
                  {
                    "bottom": 3,
                    "color": "#0D0D0D",
                    "fontSize": 31,
                    "marginRight": 16,
                  },
                ],
                {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              [
                {
                  "color": "#0D0D0D",
                  "fontFamily": "Poppins_400Regular",
                  "fontSize": 31,
                },
                {
                  "fontSize": 31,
                },
              ]
            }
          >
            Settings
          </Text>
        </View>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              [
                {
                  "color": "#0D0D0D",
                  "fontSize": 35,
                },
              ],
              {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </RNGestureHandlerButton>
    </View>
  </View>
</RCTScrollView>
`;
